ui/app/src/modules/Preview/Tools/EditFormPanel.tsx

```diff
@@ -52,6 +158,52 @@ export default function EditFormPanel() {
     return () => document.removeEventListener('keydown', handler, false);
   }, [dispatch]);
 
+  useOnMount(() => {
+    try {
+      const fieldId = item.fieldId[0];
+      let selectedId;
+      if (fieldId) {
+        selectedId = ModelHelper.extractCollectionItem(model, fieldId, item.index);
+        selectedId = (typeof selectedId === 'string' && item.index !== undefined) ? selectedId : item.modelId;
+      } else {
+        selectedId = item.modelId;
+      }
+
+      const path = ModelHelper.prop(models[selectedId], 'path');
+
+      if (path) {
+        setSrc(`${AUTHORING_BASE}/legacy/form?site=${site}&path=${path}`);
+      } else {
+        let parentPath;
+        if (model === models[selectedId]) {
+          let parentId = findParentModelId(model.craftercms.id, childrenMap, models);
+          parentPath = models[parentId].craftercms.path;
+        } else {
+          parentPath = models[model.craftercms.id].craftercms.path;
+        }
+        setSrc(`${AUTHORING_BASE}/legacy/form?site=${site}&path=${parentPath}&isHidden=true&modelId=${selectedId}`);
+      }
+
+      const callback = (e) => {
+        if (e && e.data && e.data.type === EMBEDDED_LEGACY_FORM_CLOSE) {
+          setOpen(false);
+          if (e.data.refresh) getHostToGuestBus().next({ type: RELOAD_REQUEST })
+        } else if (e && e.data && e.data.type === EMBEDDED_LEGACY_FORM_RENDERED && path) {
+          setLoading(false);
+        } else if (e && e.data && e.data.type === EMBEDDED_LEGACY_CHILD_FORM_RENDERED && !path) {
+          setLoading(false);
+        }
+      };
```

Comment:
Here, use rxjs. The `e && e.data && e.data.type` move to a filter operator and the type check moves to a switch inside a subscribe.